src/lib.rs

```diff
@@ -300,37 +300,20 @@ where
     [r * scale, g * scale, b * scale]
 }
 
-fn encode_dc(value: [f64; 3]) -> usize {
-    let rounded_r = linear_to_srgb(value[0]);
-    let rounded_g = linear_to_srgb(value[1]);
-    let rounded_b = linear_to_srgb(value[2]);
-    ((rounded_r << 16) + (rounded_g << 8) + rounded_b) as usize
+fn encode_dc([r, g, b]: [f64; 3]) -> usize {
+    let rounded = |v| linear_to_srgb(v);
+    ((rounded(r) << 16) + (rounded(g) << 8) + rounded(b)) as usize
 }
 
-fn encode_ac(value: [f64; 3], maximum_value: f64) -> usize {
-    let quant_r = f64::floor(f64::max(
-        0f64,
-        f64::min(
-            18f64,
-            f64::floor(sign_pow(value[0] / maximum_value, 0.5) * 9f64 + 9.5),
-        ),
-    ));
-    let quant_g = f64::floor(f64::max(
-        0f64,
-        f64::min(
-            18f64,
-            f64::floor(sign_pow(value[1] / maximum_value, 0.5) * 9f64 + 9.5),
-        ),
-    ));
-    let quant_b = f64::floor(f64::max(
-        0f64,
-        f64::min(
-            18f64,
-            f64::floor(sign_pow(value[2] / maximum_value, 0.5) * 9f64 + 9.5),
-        ),
-    ));
-
-    (quant_r * 19f64 * 19f64 + quant_g * 19f64 + quant_b) as usize
+fn encode_ac([r, g, b]: [f64; 3], maximum_value: f64) -> usize {
+    let quant = |v| {
+        (sign_pow(v / maximum_value, 0.5) * 9. + 9.5)
+            .floor()
+            .min(18.)
+            .max(0.)
+    };
+
+    (quant(r) * 19f64 * 19f64 + quant(g) * 19f64 + quant(b)) as usize
```

Comment:
Using a closure here drastically reduces the code size. It has the exact same behavior.

Using a closure will not have any overhead, since the virtual function does not escape the parent function.